src/Scripts/Ecs/Core/System.gd

```diff
@@ -12,4 +12,7 @@ func add(e:Entity):
 			# print("Can't add " + str(e) + " to " + str(self) + " because it lacks a component of type " + str(type))
 			return
 		
-	entities.append(e)
\ No newline at end of file
+	entities.append(e)
+
+func on_update():
+	pass
```

Comment:
I thought about adding this but didn't because I thought it isn't needed. Can you explain what this is for?

Reply:
Not all systems use `on_update` -- see `EntityMovementSystem`. Yet if it's not defined, godot throws an error. An empty method will prevent that.